t/t2219-job-manager-jobspec-default.t

```diff
@@ -0,0 +1,175 @@
+	flux mini submit /bin/true >id6
+'
+test_expect_success HAVE_JQ 'the redacted jobspec contains default duration' '
+	jm_getattr $(cat id6) jobspec >getattr6.json &&
+	jq -e ".jobspec.attributes.system.duration == 900" <getattr6.json
+'
+test_expect_success 'configure a malformed policy' '
+	cat >config/policy.toml <<-EOT &&
+	policy = 42
+	EOT
+	flux config reload
+'
+test_expect_success 'plugin reload fails with malformed policy' '
+	flux dmesg --clear &&
+	test_must_fail flux jobtap load --remove=all .jobspec-default &&
+	flux dmesg >dmesg.out &&
+	grep "Error parsing default policy" dmesg.out
+'
+test_expect_success 'configure a default queue with no [queues] entry' '
+	cat >config/policy.toml <<-EOT &&
+	[policy.jobspec.defaults.system]
+	queue = "pdebug"
+	[queues]
+	EOT
+	flux config reload
+'
+test_expect_success 'plugin reload fails with queue parse error' '
+	flux dmesg --clear &&
+	test_must_fail flux jobtap load --remove=all .jobspec-default &&
+	flux dmesg >dmesg2.out &&
+	grep "Error parsing pdebug policy" dmesg2.out
+'
+test_expect_success 'configure a queue with malformed [queues] entry' '
+	cat >config/policy.toml <<-EOT &&
+	[queues]
+	pdebug = 42
+	EOT
+	flux config reload
+'
+test_expect_success 'plugin reload fails with queue parse error' '
+	flux dmesg --clear &&
+	test_must_fail flux jobtap load --remove=all .jobspec-default &&
+	flux dmesg >dmesg3.out &&
+	grep "Error parsing pdebug queue policy" dmesg3.out
+'
+test_expect_success 'configure duration with queue override' '
+	cat >config/policy.toml <<-EOT &&
+	[policy.jobspec.defaults.system]
+	duration = 88
+	[queues.pdebug.policy.jobspec.defaults.system]
+	duration = 99
+	EOT
+	flux config reload
+'
+test_expect_success 'reload plugin' '
+	flux jobtap load --remove=all .jobspec-default
+'
+test_expect_success 'submit a job with no queue or duration' '
+	flux mini submit /bin/true >id7
+'
+test_expect_success HAVE_JQ 'the redacted jobspec contains default duration' '
+	jm_getattr $(cat id7) jobspec >getattr7.json &&
+	jq -e ".jobspec.attributes.system.duration == 88" <getattr7.json
+'
+test_expect_success 'submit a job with with queue but no duration' '
+	flux mini submit --setattr=system.queue=pdebug /bin/true >id8
+'
+test_expect_success HAVE_JQ 'the redacted jobspec contains queue duration' '
+	jm_getattr $(cat id8) jobspec >getattr8.json &&
+	jq -e ".jobspec.attributes.system.duration == 99" <getattr8.json
+'
+test_expect_success 'submit a job with with queue and duration' '
+	flux mini submit --setattr=system.queue=pdebug -t 111 /bin/true >id9
+'
+test_expect_success HAVE_JQ 'the redacted jobspec contains user duration' '
+	jm_getattr $(cat id9) jobspec >getattr9.json &&
+	jq -e ".jobspec.attributes.system.duration == 111" <getattr9.json
+'
+test_expect_failure 'submit a job to unknown queue fails' '
+	test_must_fail flux mini submit \
+		--setattr=system.queue=pbatch /bin/true 2>submit.err &&
+	grep "Error parsing pbatch queue policy" submit.err
+'
+
+test_done
```

Comment:
i think one test variant we're missing is a default queue has been specified.  So I think ...

```
[policy.jobspec.defaults.system]
queue = pbatch
[queues.pbatch.policy.jobspec.defaults.system]
duration = 123
[queues.pdebug.policy.jobspec.defaults.system]
duration = 456
```

(hopefully i did that right)

Reply:
Ah good catch.